src/Raven.Client.NodaTime/Extensions.cs

```diff
@@ -1,90 +1,94 @@
-            conventions.RegisterQueryValueConverter<LocalDateTime>(CustomQueryValueConverters.LocalDateTimeConverter);
-            conventions.RegisterQueryValueConverter<LocalDate>(CustomQueryValueConverters.LocalDateConverter);
-            conventions.RegisterQueryValueConverter<LocalTime>(CustomQueryValueConverters.LocalTimeConverter, RangeType.Long);
-            conventions.RegisterQueryValueConverter<Offset>(CustomQueryValueConverters.OffsetConverter, RangeType.Long);
-            conventions.RegisterQueryValueConverter<Duration>(CustomQueryValueConverters.DurationConverter, RangeType.Long);
-            conventions.RegisterQueryValueConverter<OffsetDateTime>(CustomQueryValueConverters.OffsetDateTimeConverter);
-            conventions.RegisterQueryValueConverter<Period>(CustomQueryValueConverters.PeriodConverter);
-            conventions.RegisterQueryValueConverter<ZonedDateTime>(CustomQueryValueConverters.ZonedDateTimeConverter);
-            conventions.RegisterQueryValueConverter<DateTimeZone>(CustomQueryValueConverters.DateTimeZoneConverter);
-
-            // Register query translators
-            conventions.RegisterCustomQueryTranslator<OffsetDateTime>(x => x.ToInstant(), CustomQueryTranslators.OffsetDateTimeToInstantTranslator);
-            conventions.RegisterCustomQueryTranslator<OffsetDateTime>(x => x.LocalDateTime, CustomQueryTranslators.OffsetDateTimeLocalDateTimeTranslator);
-            conventions.RegisterCustomQueryTranslator<ZonedDateTime>(x => x.ToInstant(), CustomQueryTranslators.ZonedDateTimeTimeToInstantTranslator);
-
-            return conventions;
-        }
-
-        public static TimeSpan ToTimeSpan(this LocalTime localTime)
-        {
-            return new TimeSpan(localTime.TickOfDay);
-        }
-
-        public static LocalTime ToLocalTime(this TimeSpan timeSpan)
-        {
-            return LocalTime.FromTicksSinceMidnight(timeSpan.Ticks);
-        }
-    }
-}
+using Raven.Client.Json.Serialization.NewtonsoftJson;
+using Raven.Client.NodaTime.JsonConverters;
+using Raven.Imports.NodaTime.Serialization.JsonNet;
+
+namespace Raven.Client.NodaTime
+{
+    public static class Extensions
+    {
+        public static T ConfigureForNodaTime<T>(this T documentStore)
+            where T : IDocumentStore
+        {
+            return documentStore.ConfigureForNodaTime(DateTimeZoneProviders.Tzdb);
+        }
+
+        public static T ConfigureForNodaTime<T>(this T documentStore, IDateTimeZoneProvider zoneProvider)
+            where T : IDocumentStore
+        {
+            documentStore.Conventions.ConfigureForNodaTime(zoneProvider);
+            return documentStore;
+        }        
+        
+        public static DocumentConventions ConfigureForNodaTime(this DocumentConventions conventions, IDateTimeZoneProvider zoneProvider)
+        {
+            if (!(conventions.Serialization is NewtonsoftJsonSerializationConventions jsonSerializationConventions))

```

Comment:
With them introducing this abstraction, I was thinking that there was maybe a plan to enable switching to a different serializer (maybe the new `System.Text.Json` stuff) in the future. And since the converters here are only for Newtonsoft.Json, I added the check.

But we could just hard-cast it to `NewtonsoftJsonSerializationConventions` and revisit in the future if neccessary.

Reply:
Oh wait, now I see it is not only a check, but also a pattern match!

Reply:
Yeah, maybe it's not the best idea to do combine it in a negated if-clause… Should I rewrite it to be more obvious?

Reply:
Nah, I think it is perfectly fine.